pkg/kubelet/pluginmanager/pluginwatcher/plugin_watcher.go

```diff
@@ -123,6 +110,12 @@ func (w *Watcher) init() error {
 // Walks through the plugin directory discover any existing plugin sockets.
 // Ignore all errors except root dir not being walkable
 func (w *Watcher) traversePluginDir(dir string) error {
+	// watch the new dir
+	err := w.fsWatcher.Add(w.path)
```

Comment:
this should be:
```
w.fsWatcher.Add(dir)
```

also, is calling `w.fsWatcher.Add` twice on the same path problematic? do we need to avoid calling `w.fsWatcher.Add(path)` when `path == dir` inside the Walk callback?

Reply:
> w.fsWatcher.Add(dir)

Nice catch... I shoud be more careful.

> is calling w.fsWatcher.Add twice on the same path problematic?

I have tested locally and it would not return an error, but according to the source code it would re-register the fd https://github.com/fsnotify/fsnotify/blob/7f4cf4dd2b522a984eaca51d1ccee54101d3414a/kqueue.go#L227, so I think we'd better avoid calling `fsWatcher.Add` twice on the same path.

Reply:
>> w.fsWatcher.Add(dir)
>
> Nice catch... I shoud be more careful.

>> is calling w.fsWatcher.Add twice on the same path problematic?
>
> I have tested locally and it would not return an error, but according to the source code it would re-register the fd

can we make sure there are tests that exercise these paths

Reply:
It is already midnight here, I would fix that tomorrow :)